File: feb/4 feb/lc62.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        mat = [[0]*m for i in range(n)]
        for i in range(n):
            for j in range(m):
                if i == 0 and j == 0:
                    mat[i][j] = 1
                elif i == 0:
                    mat[i][j] = mat[i][j-1]
                elif j == 0:
                    mat[i][j] = mat[i-1][j]
                else:
                    mat[i][j] = mat[i][j-1] + mat[i-1][j]
        return mat[n-1][m-1]

File: june/2june/lc62.java
class Solution {
    public int uniquePaths(int m, int n) {
        int[][] arr = new int[m][n];
        arr[0][0] = 1;
        for(int i=0;m>i;i++){
            for(int j = 0;n>j;j++){
                int a = 0;
                if(i != 0) a = arr[i-1][j];
                if(j != 0) a += arr[i][j-1];
                if(i == 0 && j == 0) a = 1;
                arr[i][j] = a;
            }
        }
        return arr[m-1][n-1];
    }
}